.github/workflows/linux.yaml

```diff
@@ -0,0 +1,25 @@
+name: Linux
+on: [push, pull_request]
+
+jobs:
+  linux:
+    strategy:
+      matrix:
+        ubuntu_version: [ubuntu-18.04, ubuntu-20.04]
+        compiler: [clang, gcc]
+    runs-on: ${{ matrix.ubuntu_version }}
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
+      - name: Build
+        env:
+          CC: ${{ matrix.compiler }}
+        run: |
```

Comment:
Please capture this `run` block as a distinct script, and make sure it explicitly installs all build dependencies (See my `Docker.autogen` PR maybe?) 

That script then can be called for GitHub Actions, as well as other CI systems like the existing Travis-CI, and lets us capture more architectures (and probably move towards more comprehensive testing).

Reply:
Should this script also support other distros than Debian-based ones ? Should I use the same script with needed modifications for Solaris and BSD builds ?
I'm thinking to remove Travis CI support when all these CI will be working, as they will totally replace what Travis CI actually does.

Reply:
- Other distros: in a separate PR to modify that script afterwards. BSDs & Alpine are my prime candidates, if you want Solaris/Illumos, go for it. 
- Removing Travis-CI: Not yet: GitHub Actions doesn't directly support other architectures, and Travis-CI supports ppc64 already.

Meta: we're probably going to need a more diverse build environment to test all of this, as I'm not aware of any CI service that natively supports all of the architectures & distributions we're discussing here

Reply:
I previously did a multi libc and multi arch build for PPP, I can do the same here if you want to : https://github.com/paulusmack/ppp/actions?query=workflow%3ABuildroot

Reply:
That would be fantastic! If you've got ongoing interest in `radvd`, are you interested in joining as a contributor (mostly so we have more people to review changes as well)

Reply:
Hi @robbat2,
I'm too busy for becoming a contributor, sorry.
Are you OK for a script like `test-build.sh` located in `test` directory ?
If OK, I modify this PR like you said before, then I can do separate PRs for BSD and Buildroot.

Reply:
@robbat2: Please read the @RICCIARDI-Adrien comment.

Reply:
@RICCIARDI-Adrien sorry, please do implement that `test-build.sh` script.

Reply:
Hi,
What's the blocking point ?
I pressed the "Resolve conversation" button in case this can unlock something, if you feel the PR is not complete feel free to reblock it.